From 2ba653d4b7153fbf555e7e130f1f2ac759c7d2d0 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Sun, 16 Apr 2023 11:44:28 +0300
Subject: [PATCH] lx2160acex7: SD1 protocol 8: revert dpmac link type back to
 fixed

On QSFP28 connector there is no sfp+ cage providing a linux phy object
to each dpmac instance when configured as 4 individual NICs.
This breaks the serdes driver logic for enabling and disabling serdes
and can prevent link-up.

Fixes: 6f13933: "lx2160acex7: upgrade network ports from fixed link to phy"

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 config/lx2160a/CEX7/dpc-8_x_usxgmii.dts | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/config/lx2160a/CEX7/dpc-8_x_usxgmii.dts b/config/lx2160a/CEX7/dpc-8_x_usxgmii.dts
index 7b0e576..0f6114e 100644
--- a/config/lx2160a/CEX7/dpc-8_x_usxgmii.dts
+++ b/config/lx2160a/CEX7/dpc-8_x_usxgmii.dts
@@ -72,16 +72,16 @@
 	board_info {
 		ports {
 			mac@3 {
-				link_type = "MAC_LINK_TYPE_PHY";
+				link_type = "MAC_LINK_TYPE_FIXED";
 			};
 			mac@4 {
-				link_type = "MAC_LINK_TYPE_PHY";
+				link_type = "MAC_LINK_TYPE_FIXED";
 			};
 			mac@5 {
-				link_type = "MAC_LINK_TYPE_PHY";
+				link_type = "MAC_LINK_TYPE_FIXED";
 			};
 			mac@6 {
-				link_type = "MAC_LINK_TYPE_PHY";
+				link_type = "MAC_LINK_TYPE_FIXED";
 			};
 			mac@7 {
 				link_type = "MAC_LINK_TYPE_PHY";
-- 
2.35.3

